crates/kitsune2/src/factories/mem_peer_store/test.rs

```diff
@@ -0,0 +1,283 @@
+
+    s.insert(vec![
+        AgentBuild {
+            expires_at: Some(Timestamp::now()),
+            ..Default::default()
+        }
+        .build(),
+        AgentBuild {
+            expires_at: Some(
+                Timestamp::now() + std::time::Duration::from_secs(10),
+            ),
+            ..Default::default()
+        }
+        .build(),
+    ]);
+
+    s.do_prune(
+        std::time::Instant::now(),
+        Timestamp::now() + std::time::Duration::from_secs(5),
+    );
+
+    assert_eq!(1, s.get_all().len());
+}
+
+#[test]
+fn happy_get() {
+    let mut s = Inner::new(std::time::Instant::now());
+
+    s.insert(vec![AgentBuild {
+        agent: Some(AGENT_1),
+        ..Default::default()
+    }
+    .build()]);
+
+    let a = s.get(AGENT_1).unwrap();
+    assert_eq!(a.agent, AGENT_1);
+}
+
+#[test]
+fn happy_get_all() {
+    let mut s = Inner::new(std::time::Instant::now());
+
+    s.insert(vec![
+        AgentBuild {
+            agent: Some(AGENT_1),
+            ..Default::default()
+        }
+        .build(),
+        AgentBuild {
+            agent: Some(AGENT_2),
+            ..Default::default()
+        }
+        .build(),
+    ]);
+
+    let mut a = s
+        .get_all()
+        .into_iter()
+        .map(|a| a.agent.clone())
+        .collect::<Vec<_>>();
+    a.sort();
+    assert_eq!(&[AGENT_1, AGENT_2], a.as_slice());
+}
+
+#[test]
+fn fixture_get_overlapping_storage_arc() {
+    const fn u32f(f: f64) -> u32 {
+        (u32::MAX as f64 * f) as u32
+    }
+
+    #[allow(clippy::type_complexity)]
+    const F: &[(&[&str], BasicArc, &[(&str, BasicArc)])] = &[
+        (
+            &["a", "b"],
+            BASIC_ARC_FULL,
+            &[("a", BASIC_ARC_FULL), ("b", BASIC_ARC_FULL)],
+        ),
+        (
+            &[],
+            BASIC_ARC_FULL,
+            &[("a", BASIC_ARC_EMPTY), ("b", BASIC_ARC_EMPTY)],
+        ),
+        (
+            &[],
+            BASIC_ARC_EMPTY,
+            &[("a", BASIC_ARC_FULL), ("b", BASIC_ARC_FULL)],
+        ),
+        (
+            &["a"],
+            Some((0, u32::MAX / 2)),
+            &[
+                ("a", Some((400, u32::MAX / 2 - 400))),
+                ("b", Some((u32f(0.8), u32f(0.9)))),
+            ],
+        ),
+    ];
+
+    for (exp, q, arc_list) in F.iter() {
+        let mut s = Inner::new(std::time::Instant::now());
+
+        for (arc_name, arc) in arc_list.iter() {
+            s.insert(vec![AgentBuild {
+                storage_arc: Some(*arc),
+                url: Some(Some((*arc_name).into())),
+                ..Default::default()
+            }
+            .build()]);
+        }
+
+        let mut got = s
+            .get_overlapping_storage_arc(*q)
+            .into_iter()
+            .map(|info| info.url.as_ref().unwrap().clone())
+            .collect::<Vec<_>>();
+
+        got.sort();
+
+        assert_eq!(exp, &got.as_slice());
+    }
+}
+
+#[test]
+fn fixture_get_near_location() {
```

Comment:
might be good to include zome 0 arc and expired agents and confirm they aren't included in results

Reply:
Done: https://github.com/holochain/kitsune2/pull/19/commits/24bef62698948159e9e3ca5f0f11593b38c51eb1